.circleci/config.yml

```diff
@@ -98,7 +98,14 @@ jobs:
 
     - run:
         name: Generate test app
-        command: (git diff --name-only master | grep -qG 'generators\/hyrax') || bundle exec rake engine_cart:generate
+        command: |
+          has_generator_changes=`(git diff --name-only master | grep -G 'generators\/hyrax')`
+          if [ $? == 0 ] || [ ! -f .internal_test_app/Gemfile ]; then
```

Comment:
This checks that the previous command (git diff and grepping for any generators/hyrax file) exitted with a success (it found a changed file).. or if the .internal_test_app/Gemfile does not exist (because the cache has been cleared or it hasn't yet been cached). 

then it exits with the same status code from the engine cart generate so that the build job either succeeds or fails depending on the outcome of generation.

...

Or this just exits with a successful code because there are no generator file change and the .internal_test_app/Gemfile exists already.